main.go

```diff
@@ -169,63 +170,76 @@ func main() {
 const chainLoveURL = "https://api.chain.love/"
 
 func getAnchorPoint(cctx *cli.Context) (*blockGetter, *lchtypes.TipSet, error) {
-	sourceSelect := []string{"car", "tipset-cids", "rpc-endpoint", "rpc-fullnode"}
+	sourceSelect := []string{"car", "rpc-endpoint", "rpc-fullnode"}
 
-	var isSet int
+	var countHeadSources int
 	for _, s := range sourceSelect {
 		if cctx.IsSet(s) {
-			isSet++
+			countHeadSources++
 		}
 	}
-	if isSet == 0 && cctx.Bool("trust-chainlove") {
-		isSet++
+	if countHeadSources == 0 && cctx.Bool("trust-chainlove") {
+		countHeadSources++
 		if err := cctx.Set("rpc-endpoint", chainLoveURL); err != nil {
 			return nil, nil, err
 		}
 	}
 
-	if isSet != 1 {
-		return nil, nil, xerrors.Errorf("you must specify exactly one of: %s", strings.Join(sourceSelect, ", "))
+	if countHeadSources > 1 {
+		return nil, nil, xerrors.Errorf("you can not specify more than one CurrentTipsetSource of: %s", strings.Join(sourceSelect, ", "))
+	} else if countHeadSources == 0 && !cctx.IsSet("tipset-cids") {
+		return nil, nil, xerrors.New("you have not specified any CurrentTipsetSource - you must provide the tipset excplicitly via '--tipset-cids'")
+	}
+
+	rpcAddr := cctx.String("rpc-fullnode")
+	if rpcAddr == "" {
+		rpcAddr = cctx.String("rpc-endpoint")
 	}
 
 	ctx := cctx.Context
 	var err error
 	var bg *blockGetter
-	var tsk lchtypes.TipSetKey
+	var tsk *lchtypes.TipSetKey
 	var ts *lchtypes.TipSet
 
-	if cctx.IsSet("car") {
-		bg, tsk, err = getStateFromCar(ctx, cctx.String("car"))
-		if err != nil {
-			return nil, nil, err
-		}
-	} else if cctx.IsSet("tipset-cids") {
+	// supplied TSK takes precedence
+	if cctx.IsSet("tipset-cids") {
 		cidStrs := cctx.StringSlice("tipset-cids")
 		var cids []cid.Cid
 		for _, s := range cidStrs {
-			c, err := cid.Decode(s)
-			if err != nil {
-				return nil, nil, err
+			// urfave is dumb wrt multi-value flags, do some postprocessing
```

Comment:
What's the problem, the commas aren't being respected? Seems like probably user error on our end then right?